src/org.xtuml.bp.core/src/org/xtuml/bp/core/util/BPClassLoader.java

```diff
@@ -22,35 +22,127 @@
 
 package org.xtuml.bp.core.util;
 
+import java.io.File;
+import java.io.RandomAccessFile;
+import java.io.IOException;
 import java.io.InputStream;
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Iterator;
 
 
-public class BPClassLoader {
+
+import org.xtuml.bp.core.CorePlugin;
+
+
+public class BPClassLoader extends ClassLoader {
 	
-	public BPClassLoader()
+	public BPClassLoader(String[] classpath, ClassLoader parent)
 	{
+      super( parent);
+      ClassPathEntries = new ArrayList<String>();
+      for (int i = 0; i < classpath.length; i++) {
+    	  addClassPathEntry(classpath[i]);
+      }
 	}
 
+	private ArrayList<String> ClassPathEntries;
 	
-	public void initialize() {
-	}
 	public void addClassPathEntry(String path) {
+		synchronized(ClassPathEntries) {
+			if (ClassPathEntries.indexOf(path) == -1) {
+				ClassPathEntries.add(path);
+			}
+		}
 	}
 
 	public Class<?> loadClass(String name) throws ClassNotFoundException {
-	  return null;
+      Class<?> clazz = null;

```

Comment:
The indentation in this section is really messed up.  I think you need to set the preference to "Insert spaces for tabs" and then reformat.